src/evict.c

```diff
@@ -522,20 +527,24 @@ int performEvictions(void) {
     if (!isSafeToPerformEvictions()) return EVICT_OK;
 
     int keys_freed = 0;
-    size_t mem_reported, mem_tofree;
+    size_t mem_reported, mem_tofree, mem_used;
     long long mem_freed = 0; /* Maybe become negative */
     mstime_t latency, eviction_latency;
     long long delta;
     int replicas = listLength(server.replicas);
     int result = EVICT_FAIL;
 
-    if (getMaxmemoryState(&mem_reported, NULL, &mem_tofree, NULL) == C_OK) {
+    if (getMaxmemoryState(&mem_reported, &mem_used, &mem_tofree, NULL, server.key_eviction_memory) == C_OK) {
         result = EVICT_OK;
         goto update_metrics;
     }
 
     if (server.maxmemory_policy == MAXMEMORY_NO_EVICTION || (iAmPrimary() && server.import_mode)) {
-        result = EVICT_FAIL; /* We need to free memory, but policy forbids or we are in import mode. */
+        if (mem_used >= server.maxmemory) {
+            result = EVICT_FAIL; /* We need to free memory, but policy forbids or we are in import mode. */
+        } else {
+            result = EVICT_OK; /* used_memory greater than key_eviction_memory, but not reach OOM */
+        }
         goto update_metrics;
```

Comment:
We still want to perform eviction when used_memory is greater than key_eviction_memory I think
```suggestion
            goto update_metrics;
        }
```